### PR TITLE
microtime: Add conversion API from datetime and ISO-8601

### PR DESCRIPTION
**Pull request by besfahbod**

### CODE REVIEW

**Comment by besfahbod** [comment]
@JelleZijlstra r?

File: qcore/microtime.py [hunk 1]
```diff
@@ -148,14 +159,24 @@ def format_utime_as_iso_8601(utime, *, drop_subseconds=False, tz=timezone.utc):
...
-    "2022-10-28T23:37:38+00:00"
+    "2022-10-31T18:02:03.123456+00:00"
 
     """
     if drop_subseconds:
         utime = int(utime / SECOND) * SECOND
     return utime_as_datetime(utime, tz=tz).isoformat()
 
 
+def iso_8601_as_utime(iso_datetime: str):
```
Comment (JelleZijlstra) [inline qcore/microtime.py:170]: Is Cython OK with the type annotation here?

I believe Cython has some support for type annotations now, but that may be only in the eternally-in-alpha 3.0 branch.

Comment (besfahbod) [reply qcore/microtime.py:170]: Oh, I didn't realize I added this. I didn't get any error from CY, but let me remove it for consistency.

**Comment by besfahbod** [comment]
Apparently PY3.6 doesn't support the API I'm using here.

```
  E   AttributeError: type object 'datetime.datetime' has no attribute 'fromisoformat'
```

Should I condition the new API to PY >3.6?

**Comment by besfahbod** [comment]
Actually, took a feature-check approach for enabling the API depending on the new PY feature.

File: qcore/microtime.pyi [hunk 2]
```diff
@@ -62,6 +63,10 @@ def format_utime_as_iso_8601(
     utime: Utime, *, drop_subseconds: bool = False, tz: timezone = timezone.utc
 ) -> str: ...
 
+# datetime.fromisoformat() is new in Python 3.7.
+if hasattr(datetime, "fromisoformat"):
```
Comment (JelleZijlstra) [inline qcore/microtime.pyi:67]: This isn't allowed in stubs, we must use `if sys.version_info >= (3, 7):`

Comment (besfahbod) [reply qcore/microtime.pyi:67]: Oh, right. Not sure how the checks are passing, though... :/

Btw, would I need to `import sys`, or that's not needed?

**Review by JelleZijlstra — APPROVED** [review]

**Pull request merged**